language/move-vm/runtime/src/runtime.rs

```diff
@@ -280,15 +292,27 @@ impl VMRuntime {
             .map_err(|err| err.finish(Location::Undefined))?;
 
         // run the function
-        Interpreter::entrypoint(
+        let return_vals = Interpreter::entrypoint(
             func,
             ty_args,
             signers_and_args,
             data_store,
             cost_strategy,
             &self.loader,
             log_context,
-        )
+        )?;
+
+        if !return_vals.is_empty() {
```

Comment:
This should be deleted + return values propagated--it is ok for script functions to return values, but not tx scripts.